js/npm/MITMProxy.hx

```diff
@@ -0,0 +1,154 @@
+package js.npm;
+
+import js.Error;
+import js.node.Buffer;
+import js.node.http.ClientRequest;
+import js.node.http.IncomingMessage;
+import js.node.http.ServerResponse;
+import js.node.stream.Duplex.IDuplex;
+
+typedef EmptyHandler = ProxyContext -> (Void -> Void) -> Void;
+typedef ErrorHandler = Null<ProxyContext> -> Error -> String -> Void;
+typedef DataHandler<T> = ProxyContext -> T -> (Null<Error> -> Null<T> -> Void) -> Void;
+typedef WsHandler = ProxyContext -> String -> String -> (Null<Error> -> String -> String -> Void) -> Void;
+
+typedef ProxyOptions = {
+  ?port: Int, // The port or named socket to listen on (default: 8080).
+  ?sslCaDir: String, // Path to the certificates cache directory (default: process.cwd() + '/.http-mitm-proxy')
+  ?silent: Bool, // if set to true, nothing will be written to console (default: false)
+  ?keepAlive: Bool, // enable HTTP persistent connection
+  ?timeout: Int, // The number of milliseconds of inactivity before a socket is presumed to have timed out. Defaults to no timeout.
+  ?httpAgent: js.node.http.Agent, // The http.Agent to use when making http requests. Useful for chaining proxys. (default: internal Agent)
+  ?httpsAgent: js.node.https.Agent, // The https.Agent to use when making https requests. Useful for chaining proxys. (default: internal Agent)
+  ?forceSNI: Bool, // force use of SNI by the client. Allow node-http-mitm-proxy to handle all HTTPS requests with a single internal server.
+  ?httpsPort: Int // The port or named socket for https server to listen on. (forceSNI must be enabled)
+}
+
+typedef SslOptions = {
+  keyFile: String,
+  certFile: String,
+  ?hosts: Array<String>
+}
+
+extern class MITMProxy extends ProxyEventHandler
+implements npm.Package.Require<"http-mitm-proxy", "^0.5.0"> {
```

Comment:
Also if it's call "http-mitm-proxy", we may want to follow the convention and call the module `HttpMitmProxy` or something